flex and scotch conflicts

Hello,

I was trying to install something else that through the chain wanted to install `scotch`.  I originally thought that `scotch` was failing, it had a bunch of `undefined reference to scotchyywrap` stuff going on.  The great google gods led me to [here](http://stackoverflow.com/questions/1811125/undefined-reference-to-yywrap), and I realized it was probably a `flex` conflict.  I really have no idea what these things are doing, but `scotch@6.0.3` could not install using `flex@2.6.2`.  I simply commented out the `2.6.2` version of flex (therefore defaulting to `2.6.1`) and then `scotch@6.0.3` was able to install.

If it means anything, the package that was ultimately installing both of these was `eigen` which came from a complicated `opencv` (test) installation.

If helpful, I am compiling with `gcc 5.3.1`.  Unsure if anybody else is even affected by this, but figured I'd post it in case it's helpful to somebody else somewhere :)

--- a/var/spack/repos/builtin/packages/scotch/package.py
+++ b/var/spack/repos/builtin/packages/scotch/package.py
@@ -50,7 +50,7 @@ class Scotch(Package):
     variant('metis', default=True,
             description='Build metis and parmetis wrapper libraries')
 
-    depends_on('flex', type='build')
+    depends_on('flex@:2.6.1', type='build')
     depends_on('bison', type='build')
     depends_on('mpi', when='+mpi')
     depends_on('zlib', when='+compression')

